Garden-Service/src/Garden-Service.Infrastructure/Services/GardenService.cs

```diff
@@ -14,28 +15,91 @@ public class GardenService : IGardenService {
 
 
 	public async Task<CreateGardenResponse> CreateGardenAsync(CreateGardenRequest request) {
-		var garden = new Garden(request.Name, DateTime.Now);
-		_gardenContext.Gardens.Add(garden);
-		await _gardenContext.SaveChangesAsync();
-		var createGardenResponse = new CreateGardenResponse {
-			Succeeded = true,
-			GardenId = garden.GardenId,
-			Messages = new[] {CreateGardenResponse.Message.GardenCreationSuccessfully}
-		};
+		CreateGardenResponse createGardenResponse;
+		var garden = new Garden(request.Name, DateTime.UtcNow);
+		try {
+			_gardenContext.Gardens.Add(garden);
+			await _gardenContext.SaveChangesAsync();
+			createGardenResponse = new CreateGardenResponse {
+				Succeeded = true,
+				GardenId = garden.GardenId,
+				Messages = new[] {CreateGardenResponse.Message.GardenCreationSuccessfully}
+			};
+		}
+		catch (Exception e) {
+			if (e.GetType() == typeof(DbUpdateException) || e.GetType() == typeof(DbUpdateConcurrencyException) ||
```

Comment:
Type == Type oder Type.Equals(Type) prüft nicht auf Vererbung. Um das selbe Ergebnis zu erhalten, wie bei Javas `instanceof` nutze den is-operator: `if (e is DbUpdateException || ...]`. Falls jedoch exakte Typübereinstimmung geprüft werden sollen ist der Code korrekt.